tests/test_buffering_wrapper.py

```diff
@@ -0,0 +1,100 @@
+import gym
+import numpy as np
+import pytest
+from stable_baselines.common.vec_env import DummyVecEnv
+
+from imitation.util.buffering_wrapper import BufferingWrapper
+
+
+class CountingEnv(gym.Env):
+  """At timestep `t`, has `reward == obs == t`."""
+
+  def __init__(self):
+    self.observation_space = gym.spaces.Box(low=0, high=np.inf, shape=())
+    self.action_space = gym.spaces.Box(low=0, high=np.inf, shape=())
+    self.timestep = 0
+
+  def reset(self):
+    self.timestep = 1
+    return 0
+
+  def step(self, action):
+    assert self.timestep is not None
+    assert np.array(action) in self.action_space
+    t = self.timestep
+    self.timestep += 1
+    return t, t, False, {}
+
+
+def _make_recording_venv(error_on_premature_reset: bool,
+                         ) -> BufferingWrapper:
+  venv = DummyVecEnv([CountingEnv] * 2)
+  venv = BufferingWrapper(venv, error_on_premature_reset)
+  venv.reset()
+  return venv
+
+
+def test_pop_transitions():
+  """Check pop_transitions() result for BufferWrapper.
+
+  To make things easy to check, we use a dummy environment where the observation
+  is simply the timestep.
+  """
+  venv = _make_recording_venv(True)
+  for _ in range(3):
+    expect_obs_list = []
+    expect_acts_list = []
+
+    obs = venv.reset()
+    np.testing.assert_array_equal(obs, [0, 0])
+    expect_obs_list.append(obs)
+
+    for t in range(1, 10):
+      acts = np.random.random(size=(2,))
+      expect_acts_list.append(acts)
+
+      obs, *_ = venv.step(acts)
```

Comment:
test `step_async`, `step_wait` version (if that works, just `step()` will work too)